rmq/rmqproducer/rabbitmq_producer.py

```diff
@@ -354,14 +348,13 @@ def publish_message(self, message, routing_key):
         :param str routing_key: The routing key for the message to be published
 
         """
-        if self._channel.is_open:
-            self._channel.basic_publish(self.exchange, routing_key, message, properties=pika.BasicProperties(
-                delivery_mode=2,  # make message persistent
-            ))
-        else:
-            self._LOGGER.error("Channel not open. Message %s couldn't be published. "
-                               "Will try to publish message again if channel reopens", message)
-            return
+        if not self._channel or not self._channel.is_open or not self._connection or not self._connection.is_open:
```

Comment:
if not (self._channel and self._channel.is_open and self._connection and self._connection.is_open)